lib/docker.ts

```diff
@@ -47,12 +61,29 @@ class Docker {
   }
 
   private optionsList: string[];
+  private extractProductsByFilename: ExtractProductsByFilename;
+  private staticScanSizeLimitInBytes: number;
 
-  constructor(private targetImage: string, options?: DockerOptions) {
+  constructor(
+    private targetImage: string,
+    options?: DockerOptions,
+    staticScanSizeLimit?: number,
+  ) {
     this.optionsList = Docker.createOptionsList(options);
+    this.extractProductsByFilename = {};
+    this.staticScanSizeLimitInBytes =
+      staticScanSizeLimit || STATIC_SCAN_MAX_IMAGE_SIZE_IN_BYTES;
+  }
+
+  public getTargetImage(): string {
+    return this.targetImage;
+  }
+
+  public GetStaticScanSizeLimit(): number {
+    return this.staticScanSizeLimitInBytes;
   }
 
-  public run(cmd: string, args: string[] = []) {
+  public run(cmd: string, args: string[] = []): Promise<CmdOutput> {
```

Comment:
why Promise?

Reply:
Simply because `execute` (`lib\sub-process.ts`) returns the same